src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionWithNestedFlavors.cs

```diff
@@ -70,34 +73,46 @@ public sealed override Task<IEnumerable<SuggestedActionSet>> GetActionSetsAsync(
             {
                 var extensionManager = this.Workspace.Services.GetService<IExtensionManager>();
 
-                _nestedFlavors = extensionManager.PerformFunction(
-                    Provider, CreateAllFlavors,
-                    defaultValue: ImmutableArray<SuggestedActionSet>.Empty);
+                _nestedFlavors = await extensionManager.PerformFunctionAsync(
+                    Provider, () => CreateAllFlavorsAsync(cancellationToken),
```

Comment:
this is scary that this is capturing the cancellation token.  can we make PerformFunctionAsync take in the ct and pass it along if necessary?